### ISSUE TITLE
Unpredictable GIF saving behaviour

### ISSUE THREAD

**Issue by HuskyNator** [AuburnSounds/gamut#63]
The new gif filetype seems to have issues when exporting.
I made a simpler example case to try to find why my gif wasn't exporting properly (crashing on saving).

```d
const int w = 16;
const int h = 16;
enum scale = 4;       // Fails > 4
const int frames = 1;
ubyte[4][] img = new ubyte[4][w * h];
foreach (ubyte x; 0 .. w)
	foreach (ubyte y; 0 .. h)
		img[cast(int)(y * w) + x] = [cast(ubyte)(scale * x), cast(ubyte)(scale * y), 0, 255];

ubyte[4][] gif;
foreach (i; 0 .. frames)
	gif ~= img.dup;
Image image;
image.createLayeredViewFromData(cast(void*) gif.ptr, w, h, frames, PixelType.rgba8,
	cast(int)(w * 4 * ubyte.sizeof), cast(int)(w * h * 4 * ubyte.sizeof));
image.saveToFile("test.gif");
```
Once `scaling` is set higher than 4, like 16 (my original intent), the program exist with code `-1073740940`
When run with debugging, it fails on the `realloc` at `msf_gif.d#MSF_GIF_REALLOC` (line 147).

I'd love to hear if there's a simple thing I'm not seeing.

**Comment by p0nce** [comment]
Reproduce, please wait a bit.

**Comment by p0nce** [comment]
Was solved by using slice buffers to find the segfault.
maximum frame estimation size was wrong because of the way framing works in msf_gif, interestingly the bug is from the original library msf_gif.h.
Please use Gamut v2.3.1 or later

**Comment by HuskyNator** [comment]
There's a possibility my full code has another issue along the lines of this. I'll test it out a bit first just to be sure it's not something else.

**Comment by p0nce** [comment]
Do not hesitate to submit reports!

**Comment by HuskyNator** [comment]
I believe I solved my other issue, it was a duplicate `[]` leading to incorrect pointer behaviour.
Everything seems to work now! :)

**Comment by HuskyNator** [comment]
Scratch that, saving a gif file after calling `.flipVertical()` introduces a black frame / breaks the frame somehow.
`.flipHorizontal()` however seems to work properly.
(simple insert `image.flipVertical()` right before the call to `.saveImage` in my example above ^)

**Comment by p0nce** [comment]
Nice catch! Saving with negative pitch would be reading wrong memory.
This is fixed in v2.3.3.
